packages/mdc-data-table/README.md

```diff
@@ -173,3 +187,66 @@ Mixin | Description
+
+### `MDCDataTableAdapter`
+
+Method Signature | Description
+--- | ---
+`addClassAtRowIndex(rowIndex: number, cssClasses: string) => void` | Adds a class name to row element at given row index excluding header row.
+`getAttributeAtRowIndex(rowIndex: number, attr: string) => void` | Returns attribute value of row element at given row index excluding header row.
+`getRowCount() => number` | Returns row count excluding header row.
+`getRowElements() => HTMLElement[]` | Returns array of row elements excluding header row.
+`getRowIdAtIndex(rowIndex: number) => string | null` | Returns row id of row element at given row index based on `data-row-id` attribute on row element `tr`.
+`getRowIndexByChildElement(el: Element) => number` | Returns index of row element that contains give child element.
+`getSelectedRowCount() => number` | Returns selected row count.
+`isHeaderRowCheckboxChecked() => boolean` | Returns true if header row checkbox is checked.
+`isRowsSelectable() => boolean` | Returns true if table rows are selectable.
+`notifyRowSelectionChanged(data: MDCDataTableRowSelectionChangedEventDetail) => void` | Notifies when row selection is changed.
+`notifySelectedAll() => void` | Notifies when header row is checked.
+`notifyUnselectedAll() => void` | Notifies when header row is unchecked.
+`registerHeaderRowCheckbox() => void` | Initializes header row checkbox. Destroys previous header row checkbox instance if any.
+`registerRowCheckboxes() => void` | Initializes all row checkboxes. Destroys previous row checkbox instances if any. This is usually called when row checkboxes are added or removed from table.
+`removeClassAtRowIndex(rowIndex: number, cssClasses: string) => void` | Removes class name from row element at give row index.
+`setAttributeAtRowIndex(rowIndex: number, attr: string, value: string) => void` | Sets attribute to row element at given row index.
+`setHeaderRowCheckboxChecked(checked: boolean) => void` | Sets header row checkbox checked or unchecked.
+`setHeaderRowCheckboxIndeterminate(indeterminate: boolean) => void` | Sets header row checkbox to indeterminate.
+`setRowCheckboxCheckedAtIndex(rowIndex: number, checked: boolean) => void` | Sets row checkbox to checked or unchecked at given row index.
+
+### `MDCDataTableFoundation`
+
+Method Signature | Description
+--- | ---
+`layout() => void` | Re-initializes header row checkbox and row checkboxes when selectable rows are added or removed from table.
+`getRows() => HTMLElement[]` | Returns array of row elements.
+`setSelectedRowIds(rowIds: string[]) => void` | Sets selected row ids. Overwrites previously selected rows.
+`getSelectedRowIds() => Array<string|null>` | Returns array of selected row ids.
```

Comment:
does this actually return null? I thought data-table autogens row-ids, so all rows should have an id.

Reply:
The design decision was changed to make row id optional. This'll include 'null' in the list in absence of row-id data attribute for selected row element.